### PR TITLE
Add ability to edit + delete + split mentor material parts

### PR DESCRIPTION
**Pull request by callumacrae**
## Ticket and context

Ticket: https://dfedigital.atlassian.net/browse/CPDEL-527

### Is there anything weird that code reviewer should know?
Nope, everything is straightforward.

### Review Checks
- [x] All commit messages are meaningful and true
- [x] Added enough automated tests
- [x] All forms have an `id` attribute on them

### How can someone see it it review app?
1. Click the link to review app (posted by a `github-actions` bot below)
2. Follow the steps from the ticket.

### CODE REVIEW

File: app/views/core_induction_programmes/mentor_material_parts/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<%= govuk_breadcrumbs breadcrumbs: mentor_material_breadcrumbs(current_user, @mentor_material_part.mentor_material) %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-full">
+    <h1 class="govuk-heading-l">Content change preview</h1>
+    <%# <%= govuk_link_to "Delete lesson part", lesson_part_show_delete_path(lesson_part_id: @course_lesson_part.id), button: true, class: "govuk-button--secondary"  %1> %>
```
Comment (kicferk1) [inline app/views/core_induction_programmes/mentor_material_parts/edit.html.erb:6]: Hmmm. I'm happy to add it in, although probably needs to be switched to delete mentor material parts.

**Review by kicferk1 — COMMENTED** [review]
Could we have cucumber specs for it?

File: app/views/core_induction_programmes/mentor_material_parts/edit.html.erb (hunk 1 above)
Comment (callumacrae) [reply app/views/core_induction_programmes/mentor_material_parts/edit.html.erb:6]: image

😠😠😠😠😠

Comment (kicferk1) [reply app/views/core_induction_programmes/mentor_material_parts/edit.html.erb:6]: Hehe, sorry, I just go through everything new in the morning and your PR got reviewed as part of that 😂

File: app/views/core_induction_programmes/mentor_material_parts/show_delete.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<%= govuk_breadcrumbs breadcrumbs: mentor_material_breadcrumbs(current_user, @mentor_material_part.mentor_material) %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-full">
+    <h2 class="govuk-heading-l">Confirm you would like to remove <i><%= @mentor_material_part.title %></i> from <%= @mentor_material_part.mentor_material.title %>.</h2>
```
Comment (kicferk1) [inline app/views/core_induction_programmes/mentor_material_parts/show_delete.html.erb:5]: This should be `h1`, right? Possibly in the lesson part show delete too...

File: spec/cypress/integration/CipMentorMaterials.feature [hunk 3]
```diff
@@ -19,7 +19,7 @@ Feature: Core Induction Programme mentor materials
 
   Scenario: Admins can edit mentor materials
     Given I am logged in as "admin"
-    Given mentor_material was created as "with_mentor_material_part" with id "a4dc302c-ab71-4d7b-a10a-3116a778e8d5"
+    And mentor_material was created as "with_mentor_material_part" with id "a4dc302c-ab71-4d7b-a10a-3116a778e8d5"
```
Comment (kicferk1) [inline spec/cypress/integration/CipMentorMaterials.feature:22]: Nice

**Review by kicferk1 — COMMENTED** [review]
Looks pretty good, just one accessibility related question.

File: spec/cypress/integration/CipMentorMaterials.feature (hunk 3 above)
Comment (callumacrae) [reply spec/cypress/integration/CipMentorMaterials.feature:22]: clearly the most important change in this PR

File: app/views/core_induction_programmes/mentor_material_parts/show_delete.html.erb (hunk 2 above)
Comment (callumacrae) [reply app/views/core_induction_programmes/mentor_material_parts/show_delete.html.erb:5]: yep. probably means it needs a test adding too!

File: app/views/core_induction_programmes/mentor_material_parts/show.html.erb [hunk 4]
```diff
@@ -3,7 +3,12 @@
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds">
     <% if policy(@mentor_material_part.mentor_material).edit? %>
+      <div class="govuk-button-group">
         <%= govuk_link_to "Edit mentor material", edit_mentor_material_path(@mentor_material_part.mentor_material), button: true %>
+        <%= govuk_link_to "Edit mentor material part", edit_mentor_material_part_path(@mentor_material_part), button: true %>
+        <%= govuk_link_to "Split mentor material part", mentor_material_part_split_path(mentor_material_part_id: @mentor_material_part.id), button: true %>
+        <%= govuk_link_to "Delete mentor material part", mentor_material_part_show_delete_path(mentor_material_part_id: @mentor_material_part.id), button: true, class: "govuk-button--secondary" %>
+      </div>
```
Comment (kicferk1) [inline app/views/core_induction_programmes/mentor_material_parts/show.html.erb:11]: Can we add navigation to the bottom of the page (to next/previous, like lesson parts have?)

**Review by kicferk1 — APPROVED** [review]

**Pull request merged**